starknet_py/net/base_client.py

```diff
@@ -0,0 +1,187 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Union, Optional, List
+
+from starknet_py.net.client_models import (
+    StarknetBlock,
+    BlockState,
+    Transaction,
+    TransactionReceipt,
+    ContractCode,
+    SentTransaction,
+    InvokeFunction,
+    StarknetTransaction,
+    ContractDefinition,
+    TransactionStatus,
+)
+
+
+@dataclass
+class BlockHashIdentifier:
```

Comment:
Not the best name :(

Reply:
What name would you suggest instead?

Reply:
Why do we need it?

Reply:
Dataclasses have been converted to TypeDicts now, they're still used as parameters for get_transaction

Reply:
I'm open on suggestions for how to replace it though

Reply:
It has been removed